### PR TITLE
Improve test coverage for bulk download functions

### PR DESCRIPTION
**Pull request by james-francis-MT**
# Description
Add additional unit tests for the cqc api functions

# How to test
Unit tests so no need to test in branch

# Developer checklist
- [ ] Unit test
- [ ] Linked to Trello ticket
- [ ] Documentation up to date

### CODE REVIEW

File: tests/integration/test_cqc_api_new_integration.py [hunk 1]
```diff
@@ -10,14 +10,14 @@ class TestCQCLocationAPIIntegration(unittest.TestCase):
...
-    def test_get_object_returns_location(self):
+    def test_get_object_returns_object(self):
         result = cqc.get_object(EXAMPLE_LOCATION_ID, "locations")
 
         self.assertEqual(result["locationId"], EXAMPLE_LOCATION_ID)
         self.assertEqual(result["providerId"], "1-9098203603")
         self.assertEqual(result["organisationType"], "Location")
 
-    def test_get_page_locations_returns_all_locations_for_page(self):
+    def test_get_page_object_returns_all_objectss_for_page(self):
```
Comment (luke-a-williams) [inline tests/integration/test_cqc_api_new_integration.py:20]: So minor but can we correct the spelling on this?

**Comment by luke-a-williams** [comment]
Make sure this is up to date with main too as been a few PRs recently

File: utils/cqc_api_new.py [hunk 2]
```diff
@@ -62,13 +59,7 @@ def get_all_objects(
...
         )
 
-        if stream:
-            yield page_locations
-        else:
-            all_objects.append(page_locations)
-
-    if not stream:
-        return all_objects
+        yield page_locations
```
[the remark this answers is not in the document]
Comment (james-francis-MT) [reply utils/cqc_api_new.py:62]: It isn't valid python - it always returns a generator even with the if statement

File: tests/unit/test_cqc_api_new.py [hunk 3]
```diff
@@ -45,6 +57,88 @@ def test_get_page_locations(self, mock_get_object, mock_call_api):
...
+        test_response_timeout = TestResponse(429, {})
+        test_response_success = TestResponse(200, {})
+        get_mock.side_effect = [test_response_timeout, test_response_success]
+
+        response_json = cqc.call_api(
+            "test_url", {"test": "body"}, headers_dict={"some": "header"}
+        )
+
+        sleep_mock.assert_called_once()
+        self.assertEqual(response_json, {})
```
[the remark this answers is not in the document]
Comment (james-francis-MT) [reply tests/unit/test_cqc_api_new.py:117]: Ye, I can add these tests - The 500 test will test that code but I think it is worth the 5 mins to add the extra ones

File: utils/cqc_api_new.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (james-francis-MT) [reply utils/cqc_api_new.py:62]: I tried writing a test for the stream=false case and it didn't even work

**Review by luke-a-williams — APPROVED** [review]

**Pull request merged**